_references/component-descriptor-structure.md

```diff
@@ -0,0 +1,319 @@
+
+```json
+{
+"credentials" : {
+   "fields": {
+      "oauth": {
+         "viewClass":"OAuthFieldView",
+         "label":"Twitter Account",
+         "required" : true
+      }
+   },
+   "oauth1":{
+      "consumer_key":"{{TWITTER_API_KEY}}",
+      "consumer_secret":"{{TWITTER_API_SECRET}}",
+      "request_token_uri":"https://api.twitter.com/oauth/request_token",
+      "auth_uri":"https://api.twitter.com/oauth/authorize",
+      "access_token_uri":"https://api.twitter.com/oauth/access_token"
+   }
+  }
+}
+```
+In case of OAuth1 Authentication we must use `fields` and `oauth1` objects together.
+The `fields` object defines the type of input field and the `oauth1` object provides
+the configuration.
+
+## OAuth2
+
+Specifies OAuth 2.0 specific details of the API used by the component. For more
+details about Authenticating with OAuth 2.0 please read the
+[OAuth 2.0 specification](http://tools.ietf.org/html/rfc6749).
+
+| Property Name | Type     | Required | Description |
+| :------------ | :------: | :------: | :---------- |
+| client_id   | `string` | Yes      | The Consumer Key |
+| client_secret | `string` | Yes      | The Consumer Secret |
+| auth_uri    | `string`   | Yes | URI to obtain User Authorization |
+| token_uri   | `string`   | Yes | URI to obtain a Access Token |
+| scopes      | `array` of `strings` |   No | Scopes of the access request |
+
+Here is an example of `OAuth2` implementation in [Google Spreadsheets](https://github.com/elasticio/gspreadsheets/blob/master/component.json#L18) component:
+
+```json
+"oauth2":{
+   "client_id":"{{GOOGLE_APP_ID}}",
+   "client_secret":"{{GOOGLE_APP_SECRET}}",
+   "auth_uri":"https://accounts.google.com/o/oauth2/v2/auth",
+   "token_uri":"https://www.googleapis.com/oauth2/v4/token",
+   "scopes": [
+      "https://spreadsheets.google.com/feeds"
+   ],
+   "access_type": "offline",
+   "prompt": "consent"
+}
+```
+
+Please note that the properties `access_type` and `prompt` above are specific to Google. They are not defined in the OAuth2 specification.
+
+Sometimes you will need to access values in the `oauth2` properties you
+gathered from the user using fields, as for example is done in the [Salesforce component](https://github.com/elasticio/salesforce-component/blob/master/component.json).
+The following example demonstrates how to accomplish that.
+
+```json
+"credentials" : {
+   "fields":{
+      "prodEnv" : {
+         "label":"Environment",
+         "viewClass":"SelectView",
+         "required":true,
+         "model":{
+            "test":"Sandbox",
+            "login":"Production"
+         },
+         "prompt":"Select environment"
+       },
+       "oauth":{
+          "label":"Authentication",
+           "viewClass":"OAuthFieldView",
+           "required": true
+       }
+    },
+   "oauth2":{
+      "client_id":"{{SALESFORCE_KEY}}",
+      "client_secret":"{{SALESFORCE_SECRET}}",
+      "auth_uri":"https://{{prodEnv}}.salesforce.com/services/oauth2/authorize",
```

Comment:
For some reason the `{{prodEnv}}` var does not appear in the generated HTML. The url I see is: `https://.salesforce.com/services/oauth2/authorize`

Reply:
This is because the fluid design takes anything inside the `{{ }}` as variables and tries to find a replacement for it.